compressor/management/commands/compress.py

```diff
@@ -250,7 +250,15 @@ def compress(self, log=None, **options):
                     except Exception as e:
                         raise CommandError("An error occurred during rendering %s: "
                                            "%s" % (template.template_name, smart_text(e)))
-                    offline_manifest[key] = result
+
+                    if settings.COMPRESS_OFFLINE_USE_URL_PLACEHOLDER:
+                        # Replace settings.COMPRESS_URL with a placeholder
+                        offline_manifest[key] = result.replace(
+                            settings.COMPRESS_URL, settings.COMPRESS_URL_PLACEHOLDER
+                        )
+                    else:
+                        offline_manifest[key] = result
+
```

Comment:
i was wondering, why don't you do `result = result.replace(....)` as you do in `templatetags/compress.py `? this makes this method return results that did not get written to the manifest.

looking at this i noticed that this `compress` method writes the manifest, and the `handle` method reads that manifest, merges it with others, and writes again. wouldn't it be better to not write the manifest here, just return it, and let `handle` do the writing? @mat?